### PR TITLE
Fix #1451

### LINKED ISSUE #1451: Can't kill `dune exec ...` with C-c
Starting with Dune 1.4, it looks like when running my executable with `dune exec myProgram.exe` I can't send C-c to kill it anymore

### PR DESCRIPTION
**Pull request by ghost**
By resetting the set of blocked signals in the sub-process

### CODE REVIEW

File: src/spawn.ml [hunk 1]
```diff
@@ -0,0 +1,46 @@
+module Env = struct
+  type t = string array
+
+  let of_array t = t
+end
+
+external sys_exit : int -> 'a = "caml_sys_exit"
+
+let rec file_descr_not_standard fd =
+  if (Obj.magic (fd : Unix.file_descr) : int) >= 3 then
```
Comment (nojb) [inline src/spawn.ml]: Is this supposed to be used under Windows ?

Comment (diml) [reply src/spawn.ml]: Ah, indeed. I updated the code

File: src/spawn.mli [hunk 2]
```diff
@@ -0,0 +1,17 @@
+(** Subset of https://github.com/janestreet/spawn/blob/master/src/spawn.mli *)
```
Comment (emillon) [inline src/spawn.mli]: If that's easier to maintain, we can also vendor the actual lib in `vendor/`. (that can be done in a separate PR)

**Review by emillon — APPROVED** [review]

File: src/spawn.mli (hunk 2 above)
Comment (diml) [reply src/spawn.mli]: Yh, I didn't do it yet as it has some C code so we need to be a bit careful about it

**Pull request merged**